code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -731,3 +741,47 @@ var/list/ai_verbs_default = list(
 /mob/living/silicon/ai/proc/view_photos()
 	var/obj/item/device/camera/siliconcam/ai_camera/cam = aiCamera
 	cam.view_images()
+
+// Control a tiny drone
+/mob/living/silicon/ai/proc/control_personal_drone()
+	set name = "Control Personal Drone"
+	set desc = "Take control of your own AI-bound maintenance drone."
+	set category = "Silicon Commands"
+	
+	if(aiRestorePowerRoutine)  // Cannot switch if lack of power
+		to_chat(src, SPAN_WARNING("You lack power!"))
+	else
+		go_into_drone()
+
+// Destroy AI tiny drone
+/mob/living/silicon/ai/proc/destroy_personal_drone()
+	set name = "Destroy Personal Drone"
+	set desc = "Destroy your AI-bound maintenance drone."
+	set category = "Silicon Commands"
+	
+	if(aiRestorePowerRoutine)  // Cannot switch if lack of power
+		to_chat(src, SPAN_WARNING("You lack power!"))
+	else
+		destroy_drone()
+
+/mob/living/silicon/ai/proc/pull_to_core()
+	if (bound_drone?.mind)  // If drone exists and AI is inside it
+		src.ckey = bound_drone.ckey
+		bound_drone.mind.transfer_to(src) // Pull back mind to AI core
+
+/mob/living/silicon/ai/proc/go_into_drone()
+	// Switch to drone or spawn a new one
+	if(!bound_drone)
+		testing("[world.time] . [time_destroyed] . [drone_cooldown_time]")
+		if (world.time - time_destroyed > drone_cooldown_time)
+			try_drone_spawn(src, aibound = TRUE)
+		else
+			var/remaining = (drone_cooldown_time - (world.time - time_destroyed)) / 10
+			to_chat(src, SPAN_WARNING("Security routines hardcoded into your core force you to wait [remaining] seconds before creating a new AI bound drone."))
+	else if(src?.mind)
```

Comment:
`src` should always exist, also you should be able to throw most of the `src.` out of the code

Reply:
True, I'll try to remove them 👍